master/buildbot/worker/latent.py

```diff
@@ -195,12 +195,7 @@ def buildFinished(self, sb):
 
         if not self.building:
             if self.build_wait_timeout == 0:
-                d = self.insubstantiate()
-                # try starting builds for this worker after insubstantiating;
-                # this will cause the worker to re-substantiate immediately if
-                # there are pending build requests.
-                d.addCallback(lambda _:
-                              self.botmaster.maybeStartBuildsForWorker(self.workername))
+                self._soft_disconnect()
```

Comment:
I think _soft_disconnect does too many things, this is why I prefer using insubstantiate directly